source/Ship.cpp

```diff
@@ -2077,9 +2087,29 @@ bool Ship::CanLand() const
 
 
 
-bool Ship::CannotAct() const
+bool Ship::CannotAct(ActionType actionType) const
 {
-	return (zoom != 1.f || isDisabled || hyperspaceCount || pilotError || cloak);
+	bool canActCloaked = true;
+	if(cloak)
+		switch(actionType)
+		{
+			case ActionType::AFTERBURNER:
+				canActCloaked = attributes.Get("cloaked afterburner");
```

Comment:
C++ has implicit fallthrough for case statements, so you need a `break` after you finish each case or else it will always go through every case after the actual one you want.

Reply:
I know that but somehow it slipped ha